File: app/src/main/java/igrek/todotree/dagger/DaggerIOC.java
package igrek.todotree.dagger;


import android.support.v7.app.AppCompatActivity;

public class DaggerIOC {
	
	private static FactoryComponent appComponent;
	
	private DaggerIOC() {
	}
	
	public static void init(AppCompatActivity activity) {
		appComponent = DaggerFactoryComponent.builder().factoryModule(new FactoryModule(activity))
				.build();
	}
	
	public static FactoryComponent getFactoryComponent() {
		return appComponent;
	}
	
	/**
	 * only for testing purposes
	 */
	public static void setFactoryComponent(FactoryComponent component) {
		appComponent = component;
	}
	
}

File: app/src/main/java/igrek/todotree/dagger/DaggerIoc.kt
package igrek.todotree.dagger

import androidx.appcompat.app.AppCompatActivity
import igrek.todotree.info.logger.LoggerFactory

object DaggerIoc {

    lateinit var factoryComponent: FactoryComponent

    private val logger = LoggerFactory.logger

    private var factoryModule: FactoryModule? = null

    fun init(activity: AppCompatActivity) {
        logger.info("Initializing Dagger IOC container...")
        factoryModule = FactoryModule(activity)
        factoryComponent = DaggerFactoryComponent.builder()
                .factoryModule(factoryModule)
                .build()
    }
}
